### PR TITLE
Add printf-like formatting to Text response helper

### LINKED ISSUE #22: `Text` replacement should allow printf formatting
This is the issue to do this TODO in the code:
```
  // TODO printf support so you can do new Text('You answered %d', count)
  // https://nodejs.org/docs/latest/api/util.html#util_util_format_format_args
```

This is unaffected by `Text` future design. #20

### PR DESCRIPTION
**Pull request by crccheck**
## Why are we doing this?

The text translation helps us write smaller code, but if you need to do special formatting, like "Fetched %d out of %d", you currently have to write out the whole string yourself, which can lead to very long lines, which is one of the major reasons why the `Text` translation layer exists in the first place. This adds [`util.format`](https://nodejs.org/docs/latest/api/util.html#util_util_format_format_args) to `Text` so it knows how to do this kind of formatting in a way that's consistent with `console` and printf and gettext.

closes #22

## Did you document your work?

New functionality is in the README and there are test cases for some edge cases where I wondered what would happen.

## How can someone test these changes?

Steps to manually verify the change:

Since this is in the SDK, there's only unit tests unless you `npm link` in and do a manual test. I trusted the test cases.

2. `npm i`
3. `npm t`

## What possible risks or adverse effects are there?

There is a computation cost in adding `util.format`. It may also be unnatural when `%` naturally occurs in your text

## What are the follow-up tasks?

* roll a release

## Are there any known issues?

none

## Did the test coverage decrease?

stays the same

### CODE REVIEW

File: src/responses.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // @flow
 const fs = require('fs');
+const { format } = require('util');
```
Comment (crccheck) [inline src/responses.js]: namingthings: I think `format` in the global space might be too ambiguous, but it's currently not a problem.

File: package.json [hunk 2]
```diff
@@ -52,6 +52,6 @@
     "flow-bin": "^0.41.0",
     "istanbul": "^0.4.5",
     "mocha": "^3.2.0",
-    "sinon": "^1.17.6"
+    "sinon": "^2.0.0"
```
Comment (stripethree) [inline package.json]: Nice to see that this upgrade didn't impact tests in this case.
